### PR TITLE
Add support for parsing browser css property hacks

### LINKED ISSUE #29: Property name CSS hacks for IE6/IE7 is discarded.
I'm not a big fan of CSS hacks for sure, and who cares IE6 and IE7… but I just stumbled upon this.

CSS:

```
.wtf {
  *overflow-x: hidden;
  //max-height: 110px;
  #height: 18px;
}
```

AST by gonzales-pe

```
["ruleset",
    ["selector",
        ["simpleselector",
            ["class", ["ident", "wtf"]],
            ["s", " "]]],
    ["block",
        ["s", "\n  "],
        ["declaration",
            ["property", ["ident","*overflow-x"]],
            ["propertyDelim"],
            ["s", " "],
            ["value", ["ident", "hidden"]]],
        ["declDelim"],
        ["s", "\n  "],
        ["commentSL", "max-height: 110px;"],
        ["s", "\n  "],
        ["declaration",
            ["property", ["ident", "height"]],
            ["propertyDelim"],
            ["s", " "],
            ["value",
                ["dimension",
                    ["number", "18"],
                    ["ident", "px"]]]],
        ["declDelim"],
        ["s", "\n"]]]
```

### PR DESCRIPTION
**Pull request by JakeChampion**
This is my first attempt at contributing to this project, apologies if I've missed some steps but I could not find a contribution guide.

I've had a go at adding support for parsing browser css hacks.

This should fix #29 👍

### CODE REVIEW

File: src/css/parse.js [hunk 1]
```diff
@@ -1358,6 +1358,45 @@ function getFunction() {
  * @param {number} i Token's index number
  * @return {number} Length of the identifier
  */
+function checkIdentIncludingHacks(i) {
+  const start = i;
+
+  if (i >= tokensLength) return 0;
+
+  if (tokens[i].type === TokenType.Asterisk) i++;
```
Comment (JakeChampion) [inline src/css/parse.js:1366]: I made a new function, which is 99% the same as the `checkIdent` function but adds:
```
if (tokens[i].type === TokenType.Asterisk) i++;
if (tokens[i].type === TokenType.NumberSign) i++;
```

**Comment by tonyganch** [comment]
Hi @JakeChampion! Thank you a lot for taking time to implement the fix.
May i ask you to read this comment on the topic and say what you think of it?
https://github.com/tonyganch/gonzales-pe/issues/247#issuecomment-261914946
I can't say my opinion has changed but maybe you have some usecase i should consider.

**Comment by tonyganch** [comment]
cc @bgriffith :)

**Comment by JakeChampion** [comment]
Hi @tonyganch, @bgriffith has mentioned that they think this hack should be parseable by gonzales-pe. _Source: https://github.com/tonyganch/gonzales-pe/issues/29#issuecomment-334578251._

This is parseable by scss-lint, which uses the Sass parser. I have assumed that a goal of gonzales-pe is to parse whatever sass and less are capable of parsing.

**Comment by bgriffith** [comment]
Hey - thanks for the PR! I'll take a look over the next few days. 

Re @tonyganch - While I agree the parser should stick to supporting the spec, having seen this issue come up again and again I've been swayed to think that we should support this one (but this one only) - would you be happy to include if we make a clear note of this inclusion and to potentially plan to remove at a later date?

**Comment by JakeChampion** [comment]
@bgriffith, @tonyganch, a friendly ping about this PR. The use-case I have is migrating away from scss-lint because it is the only tool that requires our developers (The Financial Times) to install and set-up a Ruby environment. If/When this PR is merged we will require only NodeJS for developers to work on projects.

**Comment by JakeChampion** [comment]
@bgriffith, @tonyganch, is there anything I can do to help get this work merged and released at all?

**Pull request closed without merging**